packages/nimble-components/src/rich-text-mention/base/models/mapping-config.ts

```diff
@@ -0,0 +1,9 @@
+/**
+ * Common state shared across Mapping Config
+ */
+export abstract class MappingConfig {
+    public constructor(
+        public readonly mentionHref: string | undefined,
+        public readonly displayName: string | undefined
```

Comment:
Should `displayName` be on the derived class in the state objects just like it is in the config element objects?

Reply:
Currently we will support only list options for all the mention types, the basic states for the list options are `mentionHref` and `displayName`. So, the base is holding both states. Is this ok to have these two states in the base class `MappingConfig` or we need to move the `displayName` to the derived class `MappingUserConfig` like below. Like to know your thoughts?

```
export class MappingUserConfig extends MappingConfig {
    public constructor(mentionHref: string | undefined, public readonly displayName: string | undefined) {
        super(mentionHref);
    }
}
```

Reply:
I thought about it some more and decided it makes sense to keep on this `MappingConfig` base class and not on `MappingUserConfig`. It will be a requirement of every mapping type to be able to set the `displayName`. 

They may each provide that differently. `MappingMentionUser` gets it via the `display-name` attribute but a new mapping might provide it with a different attribute or by computing it from multiple attributes.